odor/src/main/scala/PostgresClient.scala

```diff
@@ -189,31 +140,44 @@ class PostgresClient(val pool: PostgresConnectionPool)(implicit ec: ExecutionCon
     returnedRows
   }
 
+  @nowarn("msg=unused value of type ROW")
   def querySingleRow[PARAMS, ROW](queryFragment: Query[PARAMS, ROW], params: PARAMS = Void): Future[ROW] = async {
     val rows = await(query[PARAMS, ROW](queryFragment, params))
     if (rows.isEmpty) throw new RuntimeException("Requested single row, but got no rows.")
     rows.head
   }
 
-  val tx = new PostgresClient.Transaction(transactionSemaphore, command[Void](_))
+  val tx = new PostgresClient.Transaction(
+    transactionSemaphore,
+    command[Void](_),
+    isolationLevel = transactionIsolationLevel match {
+      case IsolationLevel.Default              => None
+      case level: IsolationLevel.ReadCommitted => Some(level)
```

Comment:
```suggestion
      // This match is exhaustive, because all other levels are subtypes of `ReadCommitted`
      case level: IsolationLevel.ReadCommitted => Some(level)
```